### ISSUE TITLE
Estimate build time with material needed

### ISSUE THREAD

**Issue by RenaeMacLeod** [alexrj/Slic3r#1766]
It would be very helpful to have Slic3r calculate an estimated build time, along with the amount of material needed, and save it to the GCode file above or below the 'filament used' comment. It would be a bonus to see this at the bottom of the program beside the 'GCode exported' message.

**Comment by lordofhyphens** [comment]
@RenaeMacLeod Yes, but frankly it's going to be inaccurate anyway and most gcode host programs (like pronterface) already do this.

**Comment by bubnikv** [comment]
Cura contains a simplified Planner code from the Marlin firmware, albeit with the hard coded acceleration values :-) Still the time estimates given by Cura are quite precise.
One can actually build a trimmed down Marlin firmware into the Slic3r just for the time estimates. In my opinion it makes sense.

**Comment by lordofhyphens** [comment]
@bubnikv I would rather not put any code from Marlin into Slic3r myself. ;)

**Comment by thiagopeixoto16** [comment]
Hi @lordofhyphens , I'm just to check what I'm doing wrong since my UI is not showing the estimated time after gcode export.
► Slic3r 1.3.1 dev 2019.02.10.2284.05a2f902.64bit
![image](https://user-images.githubusercontent.com/4227271/53736456-8cb98480-3e68-11e9-821b-edb49419147a.png)

**Comment by thiagopeixoto16** [comment]
Updating: I tried different things on dev version and noticed that it is only being displayed at **Controller** tab when printing via USB (queue frame).
Can this be moved to the Print Summary screen for those don't print via USB?

Another suggestion:
Repetier Host has a field that allows calibration of this estimated time, it is a percentage field that allows the user to manually fix the real machine print time deviation, screenshot below:
![image](https://user-images.githubusercontent.com/4227271/54034453-a5da7200-4195-11e9-9527-d99326ecacc3.png)

**Comment by lordofhyphens** [comment]
@thiagopeixoto16 move it and open a pull request :)

Should be easy to find the code that does the display.

One feature at a time though; I have been known to implement the top line feature request and ignore additional asks (closing the issue).

**Comment by thiagopeixoto16** [comment]
Thanks for you reply!

@lordofhyphens You're completely correct, in this way we can have different issues for different requests.

About your "DIY" recommendation, I'll ask someone who know coding near me, as I told you before I'm not that guy and certainly will create more problems than solutions trying to change something :)

**Comment by saintofinternet** [comment]
so is there still no way to get an Slic3r give you the print time estimate??

**Comment by dwillmore** [comment]
After slicing, read the estimated print time in the status window in the
lower right of the window?
